internal/gcsx/random_reader.go

```diff
@@ -180,6 +216,12 @@ type randomReader struct {
 	isMRDInUse bool
 
 	metricHandle common.MetricHandle
+
+	// Stores three read-states:
+	// -1 represents the 1st read, to make the gcs request size GCSFuse receives from kernel.
+	// non-negative offset represents the 2nd read, used to decide the read-pattern relative to the 1st read.
+	// -2 represents the 3rd read (or disabled state), to disable this logic.
+	randReadOptimizerState randReadOptimizerState
```

Comment:
Will this cause any performance issues for sequential reads? I think yes, and if we're only addressing this issue for random reads, why don't we directly route to the random reader implementation when the flag is true, instead of rechecking if it's a sequential read? I mean if user has pass this flag true mean they are running random read workload.